File: top_100_liked_algos_and_data_structures/maximum_product_of_two_ints_in_array/maximum_product.py
def find_max(arr):
    n = len(arr)
    if n < 2: return 

    arr.sort()
    if (arr[0] * arr[1]) > (arr[n - 1] * arr[n - 2]):
        print(f"The pair is {arr[0]} and {arr[1]}")
    else:
        print(f"The pair is {arr[n - 1]} and {arr[n - 2]}")

find_max([-10, -3, 5, 6, -20])

File: top_100_liked_algos_and_data_structures/maximum_product_of_two_ints_in_array/maximum_product.rs
fn sort(list: &mut[i32]) {
    for i in 0..list.len() {
        let mut small = i;
        for j in (i + 1)..list.len() {
            if list[j] < list[small] {
                small = j;
            }
        }

        list.swap(small, i);
    }
}

fn find_max(arr: &mut [i32]) {
    sort(arr);
    let n = arr.len();

    if (arr[0] * arr[1]) > (arr[n - 1] * arr[n - 2]) {
        println!("The pair is {} and {}", arr[0], arr[1]);
    } else {println!("The pair is {} and {}", arr[n - 1], arr[n -2])};
}

fn main() {
    let mut arr: [i32;5] = [-10, -3, 5, 6, -20];
    find_max(&mut arr);
}